docs/run/prepare/test-command.md

```diff
@@ -8,15 +8,73 @@ import TabItem from '@theme/TabItem';
 
-Charon test commands are designed to help you evaluate the performance and readiness of your candidate cluster. It allows you to test your connection to other Charon peers, the performance of your beacon node(s), and the readiness of your validator client. It prints a performance report to the standard output (which can be omitted with the `--quiet` flag) and a machine-readable TOML format of the report if the `--output-toml` flag is set.
+## Test all
 
-## Test your connection to peers
+Intended for running test across all categories. Each flag should be have a suffix of its category (i.e.: the flag `--endpoints` from the beacon tests becomes `--beacon-endpoints`). For details about each category refer to its respective section.
 
-Run tests towards other Charon peers to evaluate the effectiveness of a potential cluster setup. The command sets up a libp2p node, similarly to what Charon normally does. This test command **has to be running simultaneously with the other peers**. After the node is up it waits for other peers to get their nodes up and running, retrying the connection every 3 seconds. The libp2p node connects to relays (configurable with `p2p-relays` flag) and to other libp2p nodes via TCP. Other peer nodes are discoverable by using their ENRs. Note that for a peer to be successfully discovered, it needs to be connected to the same relay. After completion of the test suite the libp2p node stays alive (duration configurable with `keep-alive` flag) for other peers to continue testing against it. The node can be forcefully stopped as well.
+### Pre-requisites
+
+#### Running after you have started cluster creation
+
+All prerequisites for each category apply to `test all`.
+
+### Run
+
+```shell
+./charon test all \
+  --beacon-endpoints="https://ethereum-holesky-beacon-api.publicnode.com,https://ethereum-sepolia-beacon-api.publicnode.com" \
+  --mev-endpoints="https://0xac6e77dfe25ecd6110b8e780608cce0dab71fdd5ebea22a16c0205200f2f8e2e3ad3b71d3499c54ad14d6c21b41a37ae@boost-relay.flashbots.net" \
+  --peers-enrs="enr:-HW4QNDXi9MzdH9Af65g20jDfelAJ0kJhclitkYYgFziYHXhRFF6JyB_CnVnimB7VxKBGBSkHbmy-Tu8BJq8JQkfptiAgmlkgnY0iXNlY3AyNTZrMaEDBVt5pk6x0A2fjth25pjLOEE9DpqCG-BCYyvutY04TZ,enr:-HW4QO2vefLueTBEUGly5hkcpL7NWdMKWx7Nuy9f7z6XZInCbFAc0IZj6bsnmj-Wi4ElS6jNa0Mge5Rkc2WGTVemas2AgmlkgnY0iXNlY3AyNTZrMaECR9SmYQ_1HRgJmNxvh_ER2Sxx78HgKKgKaOkCROYwaDY" \
+  --output-json="/opt/charon/test/test-result.json"
+```
+
+Load test
+
+```shell
+./charon test all \
+  --beacon-endpoints="https://ethereum-holesky-beacon-api.publicnode.com,https://ethereum-sepolia-beacon-api.publicnode.com" \
+  --beacon-load-test \
+  --mev-endpoints="https://0xac6e77dfe25ecd6110b8e780608cce0dab71fdd5ebea22a16c0205200f2f8e2e3ad3b71d3499c54ad14d6c21b41a37ae@boost-relay.flashbots.net" \
+  --mev-beacon-node-endpoint="https://ethereum-beacon-api.publicnode.com" \
```

Comment:
```suggestion
  --mev-beacon-node-endpoints="https://ethereum-beacon-api.publicnode.com" \
```

Is this to get the proposer schedule to ask the relay for a block? can we just use the `--beacon-endpoints` flag if present, and do the older version of the mev test if we are given none at all?

Reply:
> Is this to get the proposer schedule to ask the relay for a block?
It is used for 2 things:
  1. Get the latest block, so we can use its hash in the next one, when we query the MEV relay (it is required by it)
  2. Fetch proposers for an epoch, so we know for which validators to query the MEV relay

> can we just use the --beacon-endpoints flag if present
Yes, we can use it and I assume you also want it for `test mev` and not only for `test all`. However, it will cause:
1. Users being unable to test rare MEV, with `test all` as it will always get stuck finding a block.
2. Misleading users as we will make absolutely no use of multiple beacon nodes.

Reply:
> Users being unable to test rare MEV, with test all as it will always get stuck finding a block.

Hmmm, because we would try the 'good mev test' and thus not test the rare relays? I guess thats a reason to not join them, but I would also consider a flag on test all like --mev-quick-test or something to do the non real way, but can't say that's necessarily easier than omitting a different flag


> Misleading users as we will make absolutely no use of multiple beacon nodes.

If we do add `s`s to `beacon-node-endpoint` flags places, we can create an issue to throw in the backlog about trying all beacon nodes with a pick list like @gsora developed for the --fallback-beacon-nodes feature, so if someone takes deep offence to the misleading name, we can point to the issue and say sorry, WIP?

Reply:
> If we do add `s`s to `beacon-node-endpoint` flags places, we can create an issue to throw in the backlog about trying all beacon nodes with a pick list like @gsora developed for the --fallback-beacon-nodes feature, so if someone takes deep offence to the misleading name, we can point to the issue and say sorry, WIP?

Okay, but... in the MEV tests we don't test or care about the BN's performance. How fast it delivers us the payloads we ask it to doesn't change the outcome of the MEV test at all, it is testing purely the relay. And that is on purpose, to remove the bias that the BN might add. That said, I'm expecting just a healthy BN with average performance to be able to answer once per slot and once per epoch on the requests I send to it.